kong/init.lua

```diff
@@ -795,8 +761,96 @@ function Kong.access()
   -- we intent to proxy, though balancer may fail on that
   ctx.KONG_PROXIED = true
 
-  if kong.ctx.core.buffered_proxying then
-    return buffered_proxy(ctx)
+  if kong.ctx.core.buffered_proxying and ngx.req.http_version() < 2 then
```

Comment:
Now we also have this requirement, on above we just set the `kong.ctx.core.buffered_proxying = true` without checking this second condition. I think the `kong.ctx.core.buffered_proxying = true` should probably reflect the fact of what really happened.

Reply:
sounds reasonable.  will check.